templates/zerver/register.html

```diff
@@ -206,6 +206,10 @@
 });
 
 $("#timezone").val(moment.tz.guess());
+
+$('#id_team_subdomain').on('input', function (e) {
```

Comment:
I think we can do something slightly nicer, which is 
* replace upper case with lower case
* replace space and _ with -
* if the user types anything else (or starts with a -), show an error message